src/LibHdb++Cassandra.cpp

```diff
@@ -276,214 +407,197 @@ bool HdbPPCassandra::find_attr_id_and_ttl(string fqdn_attr_name, CassUuid & ID,
-			cout << __func__<< ": SUCCESS in query: " << query_str.str() << endl;
-#endif
-            // TODO Improve error handling
-			const CassRow* row = cass_iterator_get_row(iterator);
-			cass_value_get_uuid(cass_row_get_column(row, 0), &ID);
-			const char * db_type_res;
-			size_t db_type_res_length;
-			cass_value_get_string(cass_row_get_column(row, 1), &db_type_res, &db_type_res_length);
-			db_type = string(db_type_res,db_type_res_length);
-			cass_int32_t ttl_val = 0;
-			if(cass_value_get_int32(cass_row_get_column(row,2), &ttl_val) == CASS_ERROR_LIB_NULL_VALUE)
-				conf_ttl = 0;
-			else
-				conf_ttl = (unsigned int) ttl_val;
-			found = true;
-    	}
-		cass_result_free(result);
-		cass_iterator_free(iterator);
+		DEBUG_MSG("SUCCESS in query: " << query_str.str() << endl);
+
+		// TODO Improve error handling
+		const CassRow* row = cass_iterator_get_row(iterator);
+		cass_value_get_uuid(cass_row_get_column(row, 0), &ID);
+		const char * db_type_res;
+		size_t db_type_res_length;
+		cass_value_get_string(cass_row_get_column(row, 1), &db_type_res, &db_type_res_length);
+		db_type = string(db_type_res,db_type_res_length);
+		cass_int32_t ttl_val = 0;
+
+		if(cass_value_get_int32(cass_row_get_column(row,2), &ttl_val) == CASS_ERROR_LIB_NULL_VALUE)
+			conf_ttl = 0;
+		else
+			conf_ttl = (unsigned int) ttl_val;
+
+		found = true;
 	}
 
+	cass_result_free(result);
+	cass_iterator_free(iterator);
 	cass_future_free(future);
 	cass_statement_free(statement);
+
 	if(!found)
 	{
-#ifdef _CASS_LIB_DEBUG
-		cout << __func__<< ": NO RESULT in query: " << query_str.str() << endl;
-#endif
-		return -1;
+		DEBUG_MSG("NO RESULT in query: " << query_str.str() << endl);
+		TRACE_MSG("Leaving");
+		return AttrNotFound;
 	}
-	if(db_type != attr_type)
+	else if(db_type != attr_type)
 	{
-#ifdef _CASS_LIB_DEBUG
-		cout << __func__<< ": FOUND ID for " << facility << "/" << attr << " but different type: attr_type="<<attr_type<<"-db_type="<<db_type << endl;
-#endif
-		return -2;
-	}
-	else
-	{
-#ifdef _CASS_LIB_DEBUG
-		cout << __func__<< ": FOUND ID for " << facility << "/" << attr << " with SAME type: attr_type="<<attr_type<<"-db_type="<<db_type << endl;
-#endif
-		return 0;
+		DEBUG_MSG("FOUND ID for " << facility << "/" << attr << " but different type: attr_type="<<attr_type<<"-db_type="<<db_type << endl);
+		TRACE_MSG("Leaving");
+		return FoundAttrWithDifferentType;
 	}
+
+	DEBUG_MSG("FOUND ID for " << facility << "/" << attr << " with SAME type: attr_type="<<attr_type<<"-db_type="<<db_type << endl);
+	TRACE_MSG("Leaving");
+	return FoundAttrWithSameType;
 }
 
 void HdbPPCassandra::extract_and_bind_bool(CassStatement* statement,
-                                           int & param_index,
-                                           int data_format /*SCALAR, SPECTRUM, ..*/,
-                                           Tango::EventData *data,
-                                           enum extract_t extract_type)
+										   int & param_index,
```

Comment:
I am planning to convert all tabs to spaces in my next pull request. This will only change formatting and documentation, not code, so to keep any important code changes out of the changeset .